### PR TITLE
[WIP][ISSUE-111] 🔀 Merge doc and requirements fixes into main

### LINKED ISSUE #111: PyYAML dependency installation fails
When installing the requirements this [issue is raised](https://github.com/yaml/pyyaml/issues/724) and it is related to [this open issue](https://github.com/yaml/pyyaml/issues/601).
The fix for me was to install PyYAML~=5.3.1, i still didn't run the project though so i don't know whether this cause side effects or not.
FYI, i am running python 3.10.10 within a virtualenv.

### PR DESCRIPTION
**Pull request by MokhFn**
All Submissions:

* [x] Have you followed the guidelines in our [Contributing documentation](https://blockchain-technology-lab.github.io/consensus-decentralization/contribute)?
* [x] Have you verified that there aren't any other open Pull Requests for the same update/change?
* [x] Does the Pull Request pass all tests?

# Description

- Updating the doc for more readability and details.
- Updating requirements to fix #111

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ subsequent levels of decentralization. Please refer to the project's
 [documentation pages](https://blockchain-technology-lab.github.io/consensus-decentralization/) for details on its architecture,
 required input, produced output, and more.
 
-Currently, the supported blockchains are:
+Currently, the supported blockchains are :
```
Comment (LadyChristina) [inline README.md:8]: I don't think it's common to have a space before a colon, so could you please remove the added spaces?

File: requirements.txt [hunk 2]
```diff
@@ -3,7 +3,7 @@ pytest>=6.2.5
 coverage==7.1.0
 flake8==6.0.0
 google-cloud-bigquery~=3.7.0
-PyYAML~=5.4
+PyYAML~=5.3.1
```
Comment (LadyChristina) [inline requirements.txt:6]: Perhaps adding `>=` instead of `~=` would be better here? Does it work for you in this case?

Comment (MokhFn) [reply requirements.txt:6]: Indeed, that worked fine, updated accordingly.

File: README.md [hunk 3]
```diff
@@ -23,9 +24,9 @@ To install the tool, simply clone this project:
     git clone https://github.com/Blockchain-Technology-Lab/consensus-decentralization.git
 
 The [requirements file](requirements.txt) lists the dependencies of the project.
-Make sure you have all of them installed before running the scripts. To install
-all of them in one go, run the following command from the root directory of the
-project:
+Make sure you have all of them installed (preferably within a [`virtualenv`](https://docs.python.org/3/library/venv.html)) 
+before running the scripts. To install all of them in one go, 
+run the following commands from the root directory of the project:
```
[the remark this answers is not in the document]
Comment (MokhFn) [reply README.md]: Thanks for pointing that out, i must have missed when i rolled back some changes (i originally wanted to include commands for venv creation and activation, but then i just redirected to the docs)
It's fixed now

**Review by dimkarakostas — CHANGES_REQUESTED** [review]
Please squash the commits so that we keep a clean history. I'd suggest one commit for the docs changes and one for the requirements. Other than that LGTM

**Pull request closed without merging**